### DISCUSSION TITLE
Help beginner validate the approach with AlpineJS.

### DISCUSSION THREAD

**Discussion by grandfso** [alpinejs/alpine#2010] [1. Help] [answered]
Hello, I am new to JS and Alpine and modern frontend webdev altogether. Trying to learn the ropes.

I am trying to add Alpine to a tailwind navbar. The starting point, was the HTML code for Tailwind responsive navbar taken from https://tailwindui.com/components/application-ui/navigation/navbars

I got the mobile menu and the profile dropdown working, but have some doubts. 
Namely, I am using two variables for tracking mobile menu open status. The first one `'open'` I am using for toggling the mobile menu visibility. The second one `'isOpen'`is used for switching the displayed icon (menu icon vs. cross icon) when the menu is open/closed. 

Here's what I've got so far:
[https://0bin.net/paste/E7LHexbo#nBuQt-vSi9CBRJU3mem+iuZc+zmIxPhF+1XMKg509kV](https://0bin.net/paste/E7LHexbo#nBuQt-vSi9CBRJU3mem+iuZc+zmIxPhF+1XMKg509kV)

Given how the example from tailwind is structured I was unable to figure out a way of doing this using one variable.

Do you see a better approach avoiding two Alpine variables without changing the DIV structure?

**Comment by danddanddand** [comment, marked as answer]
You probably want to put your state on the root element and if you have to make a root div for this I wouldn’t stress about it as it’s common with Alpine to do this. You could also reach for global state using store but this doesn’t seem necessary for your example.
